app/views/smart_proxies/plugins/_puppet.html.erb

```diff
@@ -0,0 +1,29 @@
+<% dashboard = Dashboard::Data.new("puppetmaster = #{@smart_proxy.name}") %>
+<% @hosts = dashboard.hosts %>
+<% @report = dashboard.report %>
+<h3><%= _("Puppet") %></h3>
+<%= show_feature_version('puppet') %>
+<div class="col-md-2"><%= _('Hosts managed:') %></div>
+<div class="col-md-10">
+  <%= link_to @hosts.length, hosts_path(:search => "puppetmaster = \"#{@smart_proxy.name}\"") %>
+</div>
+<ul id="proxy-puppet-tab" class="nav nav-tabs nav-tabs-pf">
```

Comment:
I'm not how to handle this, I think this looks better and is very intuitive to use, where a long page with headers will look worse in my opinion (and would not work for the puppetca anyways because of the two tables of changing sizes.).